SimulationConditions/inc/BookkeeperMaker.hh

```diff
@@ -0,0 +1,43 @@
+#ifndef SimulationConditions_BookkeeperMaker_hh
+#define SimulationConditions_BookkeeperMaker_hh
+
+//
+// Makes the Bookkeeper ProditionsEntitiy
+//
+
+#include "SimulationConditions/inc/Bookkeeper.hh"
+#include "SimulationConfig/inc/BookkeeperConfig.hh"
+#include "DbTables/inc/SimEfficiencies.hh"
+
+namespace mu2e {
+
+  class BookkeeperMaker {
+  public:
+    BookkeeperMaker(BookkeeperConfig const& config):_config(config) {}
+
+    typename Bookkeeper::ptr_t fromFcl() {
+      auto ptr = std::make_shared<Bookkeeper>();
+      for (const auto& i_effConf : _config.simStageEfficiencies()) {
+	ptr->addEff(i_effConf.tag(), i_effConf.eff());
+      }
+      return ptr;
+    }
+
+    typename Bookkeeper::ptr_t fromDb(typename SimEfficiencies::cptr_t effDb) {
+      // fill the Bookkeeper with initial values
+      auto ptr = fromFcl();
+      // now overwrite with values from database
```

Comment:
I do not see a use case for providing a value in FCL and then overriding it with a DB record, as the present code does.  It would make more sense to load DB info first, and allow to override a DB value from FCL.

Reply:
> I do not see a use case for providing a value in FCL and then overriding it with a DB record, as the present code does. It would make more sense to load DB info first, and allow to override a DB value from FCL.

 This style was in the context of writing all your proditions code first as filled from fcl.  At a later step, you start overriding  pieces from the database as you develop the database context.  During this process, you have the option to turn off the database at any time and get good fcl defaults.  If you were in the situation where well-developed and reading from the database regularly, you can supply temporary overrides of the database result by the text file method.

But this raises another issue that was not fully discussed.  Currently Proditions in TrackerConditions comes 90% from fcl and 10% (potentially) from the database.  Should there be a rule or assumption that all Proditions inputs will eventually come the database or does it make sense to allow some to remain in fcl?  Are some parameters more like reco configuration rather than describing the data?

Reply:
We are working towards a majority of tracker proditions coming from the database for MDC2020.  The distinction between configuration (for simulation) and conditions (for reconstruction) is blurry.  If we imagine simulating different run periods with different configurations, then all the 'configurations' need to be conditions, so that they can be treated coherently in sim and reco.